### ISSUE TITLE
[device-port] [surnia] Motorola Moto E 2015 LTE

### ISSUE THREAD

**Issue by impreza233** [Halium/projectmanagement#97]
Tree: halium-7.1

- [x] Create manifest https://github.com/Halium/halium-devices/pull/22
- [x] Boot image and system image build successfully
- [x] Device boots into rootfs, `usb: Manufacturer: GNU/Linux Device` appears in `dmesg` on host.
- [x] LXC container starts and does not crash
- [x] libhybris tests
  * [x] test_gps
  * [x] test_hwcomposer
  * [x] test_lights
  * [x] test_vibrator
  * [x] test_wifi
  * [x] test_sensors
  * [x] test_audio
  * [x] test_camera
  * [x] test_input
  * [x] test_nfc
  * [x] test_recorder

**Comment by impreza233** [comment]
Device boots perfectly into rootfs, but it hangs on SSH connection and reboots (17/02/2019)

**Comment by impreza233** [comment]
I have written a new defconfig with the support of Stefano Melchior and his porting of Aquaris U Plus. 

[defconfig surnia v1.txt](https://github.com/Halium/projectmanagement/files/3134807/defconfig.surnia.v1.txt)

**Comment by ab1jx** [comment]
Interesting, what model exactly?  I have 3 of XT1527, which is a 2015 (I think) LTE but there were a few of them.  More into Debian than Ubuntu but I'm tempted to try it.

**Comment by impreza233** [comment]
I have the XT1524, but it should work on XT1526 and XT1527. Don't hesitate
to try on your Debian machine, because Ubuntu and Debian share a lot of
packages.

**Comment by ab1jx** [comment]
A couple questions: The https://docs.halium.org/en/latest/porting/first-steps.html page says you need 16 GB of storage (the phone has 8) but can an external SD card be used as part of that?  Android's picky about what can go on the SD, without good reasons I think.

This seems to be mostly set up for building under amd64 and i386 but I mostly use Raspberry Pis which are already armhf, do I still need to cross compile?   Hopefully this is less reliant on Java than Android.

**Comment by impreza233** [comment]
I have had success building it on an X86-64 machine with Halium official
instructions. And, also, the Halium porting + UBPorts rootfs fits in the
/data partition without any hitch (less than 4 GB of space used in that
partition).

**Comment by ab1jx** [comment]
Oh, OK, just found your XDA post at https://forum.xda-developers.com/moto-e-2015/development/wip-advances-porting-halium-ubuntu-t3848408

Usually "no route to host" means you need to set up a gateway like
route add default gw somehost
somehost can also be an IP I think. You should only have to do it once, it 
gets saved in the routing table. See man route

Usually the phone will send out a DHCP request and any DHCP server (like on a wifi AP) that hears it will send back a packet of info that sets up the gateway among other things.  You might be able to ping the AP's IP address from the phone before that, maybe not.  The phone in some sense won't have an IP of its own until the AP assigns it one.

**Comment by impreza233** [comment]
Yes, it's a scratchpad where I post some images and I do some testing with
users. If you need, don't hesitate to say me something ;)

**Comment by impreza233** [comment]
Thanks to bartbart2003 (https://github.com/bartbart2003) and Flohack, 
![lqtuXU8](https://user-images.githubusercontent.com/25793051/88417756-c46cf780-cde2-11ea-83c1-69e0d08bc0a7.jpg)
Halium has booted into Ubuntu Touch. 
Touchscreen is ok
RIL (data) is ok
WiFi, sound and calls are not working at that moment.

**Comment by impreza233** [comment]
All tests passed, and an Alpha build is being submitted to the UBPorts Forum. Thanks to @bartbart2003 for its help!!
